lib/runner/reporters/junit.xml.ejs

```diff
@@ -8,16 +8,18 @@
     tests="<%= module.tests %>" time="<%= module.time %>" timestamp="<%= module.timestamp %>">
   <% for (var item in module.completed) {
     var testcase = module.completed[item];
-    var assertions = testcase.assertions %>
+    var assertions = testcase.assertions;
+    var errorMessages = testcase.errmessages || [];
+  %>
     <testcase name="<%= item %>" classname="<%= className %>" time="<%= testcase.time %>" assertions="<%= assertions.length %>"><%
       for (var i = 0; i < assertions.length; i++) { %><% if (assertions[i].failure) { %>  <failure message="<%= assertions[i].message %>"><%= assertions[i].stackTrace %></failure><% } %>
 <% if (assertions[i].screenshots && assertions[i].screenshots.length > 0) { %><system-out><% for (var j = 0; j < assertions[i].screenshots.length; j++) { %>[[ATTACHMENT|<%= assertions[i].screenshots[j] %>]]<% } %></system-out><% } %>
     <% }
     if (testcase.failed > 0 && testcase.stackTrace) {
     %><failure message="<%= testcase.message %>"><%= testcase.stackTrace %></failure><% }
-    if (testcase.errors > 0 && testcase.errorStackTrace) {
-    %>
-    <error message="Error executing test code"><%= testcase.errorStackTrace %></error><% } %>
+    for (var i = 0; i < errorMessages.length; i++) { %>
```

Comment:
According to the schema, you can have at most 1 element (note that you cannot have both error and failure too, so should be some kind of else if...)

Reply:
if I right understand the xsd schema, there are no any limitation how many errors and failures could occur.
<xs:choice minOccurs="0">
  <xs:element name="error">...</xs:element>
  <xs:element name="failure"></xs:element>
</xs:choice>
I read that schema as "there could be 0 or more occurrences of elements, that appear in choice element. That means there could be any number of failure and errors.
Did I misunderstand the schema?

Reply:
Please disregard my comment. I thought that maxoccurs is defaulted to unbounded.

Reply:
I altered a code to generate a single error element only if failure does not exist